src/util/system.h

```diff
@@ -34,6 +35,29 @@
 #include <utility>
 #include <vector>
 
+#if defined(ENABLE_EXTERNAL_SIGNER) && defined(BOOST_POSIX_API)
+#include <fcntl.h>
+#ifdef FD_CLOEXEC
+#include <unistd.h>
+#if defined(WIN32) && !defined(__kernel_entry)
```

Comment:
Would it make sense to move all Windows related stuff from this PR to #25111?

Reply:
Isn't `defined(WIN32)` always false on the systems that we support boost::process for?